Userland/Libraries/LibCore/SocketAddress.h

```diff
@@ -90,6 +120,7 @@ class SocketAddress {
 private:
     Type m_type { Type::Invalid };
     IPv4Address m_ipv4_address;
+    IPv6Address m_ipv6_address;
```

Comment:
pretty sure it's an either or relation between these two,
so (not mandatory, if not not-advisable for now, maintainers choice) a union could be used here

Reply:
hmm, IPv6 is 128bit address and IPv6 is just 32bit address

Reply:
Union rather than a Variant?

Reply:
Well it already has a tag, so union, as Variant is just a tagged union with compiler heartattack

Reply:
@Hendiadyoin1 @circl-lastname what you suggest to do?

Reply:
Hendyadyoin suggests using a Union so that there can be only IPv4Address or IPv6Adress but not both at the same time.

Reply:
@circl-lastname sorry, I don't know how to use unions, can you provide me a guide or example?